keps/sig-storage/177-volume-snapshot/README.md

```diff
@@ -231,24 +251,288 @@ Here are the changes since the original design proposal:
+  * snapshot_operation_count
+  * snapshot_csi_total_seconds
+* Tighten the CRD schema validation to enforce immutability. See details in "keps/sig-storage/1900-volume-snapshot-validation-webhook".
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/20190731-production-readiness-review-process.md.
+
+The production readiness review questionnaire must be completed for features in
+v1.19 or later, but is non-blocking at this time. That is, approval is not
+required in order to be in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+
+-->
+
+### Feature Enablement and Rollback
+
+_This section must be completed when targeting alpha to a release._
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [x] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: VolumeSnapshotDataSource
+    - Components depending on the feature gate: kube-apiserver
+  - [ ] Other
+    - Describe the mechanism:
+    - Will enabling / disabling the feature require downtime of the control
+      plane?
+    - Will enabling / disabling the feature require downtime or reprovisioning
+      of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+
+* **Does enabling the feature change any default behavior?**
+  Yes. Enabling the feature gate will enable the volume snapshot support and disabling the feature gate will disable the volume snapshot support.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+  Yes. If the feature is disabled after it has been enabled, existing snapshots will still be there but they cannot be used as data source to provision new PVCs.
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+  If we reenable the feature that was previously rolled back, the existing snapshots can be used again as data source to provision new PVCs.
+
+* **Are there any tests for feature enablement/disablement?**
+  The e2e framework does not currently support enabling or disabling feature
+  gates. However, unit tests in each component dealing with managing data, created
+  with and without the feature, are necessary. At the very least, think about
```

Comment:
Do you have a link to where types containing the necessary data for using the source as a snapshot are shown to be ignored in a unit test?  That's what this section is talking about.  It basically checks to be make sure that data doesn't get "stuck".

Reply:
This PR added a unit test for this.  Can you check this?  https://github.com/kubernetes/kubernetes/pull/72666

Reply:
> This PR added a unit test for this. Can you check this? [kubernetes/kubernetes#72666](https://github.com/kubernetes/kubernetes/pull/72666)

that looks good.  Can you indicate that you created unit tests and perhaps provide that same link?

Reply:
Sure.